src/main/java/net/mcreator/workspace/WorkspaceFolderManager.java

```diff
@@ -45,28 +46,37 @@ public WorkspaceFolderManager(@Nonnull File workspaceFile, @Nonnull Workspace wo
 		this.workspace = workspace;
 	}
 
-	public File getWorkspaceFolder() {
-		return workspaceFolder;
-	}
+	private static List<File> listPNGsInDir(@Nullable File dir) {
+		if (dir == null)
+			return Collections.emptyList();
 
-	public ImageIcon getBlockImageIcon(String textureIdentifier) {
-		return new ImageIcon(getBlockTextureFile(textureIdentifier).getAbsolutePath());
+		List<File> retval = new ArrayList<>();
+		File[] block = dir.listFiles();
+		for (File element : block != null ? block : new File[0])
+			if (element.getName().endsWith(".png"))
+				retval.add(element);
+		return retval;
 	}
 
-	public ImageIcon getItemImageIcon(String textureIdentifier) {
-		return new ImageIcon(getItemTextureFile(textureIdentifier).getAbsolutePath());
+	public static File getSuggestedWorkspaceFoldersRoot() {
+		File workspacesFolder = new File(System.getProperty("user.home"), "MCreatorWorkspaces");
+		if (!workspacesFolder.getAbsolutePath().matches("[a-zA-Z0-9_/+\\-\\\\:()\\[\\].,@$=`' ]+")) {
+			if (OS.getOS() == OS.WINDOWS)
+				workspacesFolder = new File("C:/", "MCreatorWorkspaces");
+		}
+		return workspacesFolder;
 	}
 
-	public File getBlockTextureFile(String textureIdentifier) {
-		return new File(getBlocksTexturesDir(), textureIdentifier + ".png");
+	public File getWorkspaceFolder() {
+		return workspaceFolder;
 	}
 
-	public File getItemTextureFile(String textureIdentifier) {
-		return new File(getItemsTexturesDir(), textureIdentifier + ".png");
+	public ImageIcon getTextureImageIconFromSectionType(String textureIdentifier, TextureSection section) {
+		return new ImageIcon(getTextureFileFromSectionType(textureIdentifier, section).getAbsolutePath());
 	}
 
-	public File getOtherTextureFile(String textureIdentifier) {
-		return new File(getOtherTexturesDir(), textureIdentifier + ".png");
+	public File getTextureFileFromSectionType(String textureIdentifier, TextureSection section) {

```

Comment:
Either remove `type` word from those methods' names, or rename `TextureSection` to `TextureSectionType` so that they match each other.